src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -42,17 +45,29 @@ class HttpKernel implements HttpKernelInterface, TerminableInterface
     /**
      * Constructor.
      *
-     * @param EventDispatcherInterface    $dispatcher   An EventDispatcherInterface instance
-     * @param ControllerResolverInterface $resolver     A ControllerResolverInterface instance
-     * @param RequestStack                $requestStack A stack for master/sub requests
+     * @param EventDispatcherInterface    $dispatcher              An EventDispatcherInterface instance
+     * @param ControllerResolverInterface $resolver                A ControllerResolverInterface instance
+     * @param RequestStack                $requestStack            A stack for master/sub requests
+     * @param ArgumentResolverManager     $argumentResolverManager A manager to resolve the controller arguments
      *
      * @api
      */
-    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null)
+    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null, ArgumentResolverManager $argumentResolverManager = null)
     {
         $this->dispatcher = $dispatcher;
         $this->resolver = $resolver;
         $this->requestStack = $requestStack ?: new RequestStack();
+
+        if (null === $argumentResolverManager) {
+            $argumentResolverManager = new ArgumentResolverManager(array(
+                new RequestArgumentResolver(),
+                new RequestAttributesArgumentResolver(),
+            ));
+        }
+
+        if (method_exists($resolver, 'setArgumentResolverManager')) {
```

Comment:
If i create a ControllerResolver and use the method `setArgumentResolverManager` on it to set the manager directly on the resolver, and i then do not give HttpKernel an instance of that manager, the HttpKernel class will set it and overwrite the correct bootstrapping that was done.

Is there a purpose for having this code here?